Server/Timer.cs

```diff
@@ -364,39 +364,34 @@ public void TimerMain()
 			}
 		}
 
-		private static Queue<Timer> m_Queue = new Queue<Timer>();
+		private static ConcurrentQueue<Timer> m_Queue = new ConcurrentQueue<Timer>();
 		private static int m_BreakCount = 20000;
 
 		public static int BreakCount{ get{ return m_BreakCount; } set{ m_BreakCount = value; } }
 
-		private static int m_QueueCountAtSlice;
-
 		private bool m_Queued;
 
 		public static void Slice()
 		{
-			lock ( m_Queue )
-			{
-				m_QueueCountAtSlice = m_Queue.Count;
+			int count = 0;
 
-				int index = 0;
+			while (count++ < m_BreakCount) {
 
-				while ( index < m_BreakCount && m_Queue.Count != 0 )
-				{
-					Timer t = m_Queue.Dequeue();
-					TimerProfile prof = t.GetProfile();
+				if (!m_Queue.TryDequeue(out Timer t)) {

```

Comment:
I think the CI complained about this line.  Maybe it should be:
`if (!m_Queue.TryDequeue(out t)) {`